src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -773,269 +773,295 @@ static HtmlEntities()
+                    [ToUInt64Key("cup")] = '\x222a',
+                    [ToUInt64Key("int")] = '\x222b',
+                    [ToUInt64Key("there4")] = '\x2234',
+                    [ToUInt64Key("sim")] = '\x223c',
+                    [ToUInt64Key("cong")] = '\x2245',
+                    [ToUInt64Key("asymp")] = '\x2248',
+                    [ToUInt64Key("ne")] = '\x2260',
+                    [ToUInt64Key("equiv")] = '\x2261',
+                    [ToUInt64Key("le")] = '\x2264',
+                    [ToUInt64Key("ge")] = '\x2265',
+                    [ToUInt64Key("sub")] = '\x2282',
+                    [ToUInt64Key("sup")] = '\x2283',
+                    [ToUInt64Key("nsub")] = '\x2284',
+                    [ToUInt64Key("sube")] = '\x2286',
+                    [ToUInt64Key("supe")] = '\x2287',
+                    [ToUInt64Key("oplus")] = '\x2295',
+                    [ToUInt64Key("otimes")] = '\x2297',
+                    [ToUInt64Key("perp")] = '\x22a5',
+                    [ToUInt64Key("sdot")] = '\x22c5',
+                    [ToUInt64Key("lceil")] = '\x2308',
+                    [ToUInt64Key("rceil")] = '\x2309',
+                    [ToUInt64Key("lfloor")] = '\x230a',
+                    [ToUInt64Key("rfloor")] = '\x230b',
+                    [ToUInt64Key("lang")] = '\x2329',
+                    [ToUInt64Key("rang")] = '\x232a',
+                    [ToUInt64Key("loz")] = '\x25ca',
+                    [ToUInt64Key("spades")] = '\x2660',
+                    [ToUInt64Key("clubs")] = '\x2663',
+                    [ToUInt64Key("hearts")] = '\x2665',
+                    [ToUInt64Key("diams")] = '\x2666',
                 };
 
-            public static char Lookup(string entity)
+            public static char Lookup(ReadOnlySpan<char> entity)
             {
-                char theChar;
-                s_lookupTable.TryGetValue(entity, out theChar);
-                return theChar;
+                // To avoid an allocation, keys of type "ulong" are used in the lookup table.
+                // Since all entity strings comprise 8 characters or less and are ASCII-only, they "fit" into an ulong (8 bytes).
+                if (entity.Length <= 8)
+                {
+                    s_lookupTable.TryGetValue(ToUInt64Key(entity), out char result);
+                    return result;
+                }
+                else
+                {
+                    // Currently, there are no entities that are longer than 8 characters.
+                    return default;
+                } 
+            }
+
+            private static ulong ToUInt64Key(ReadOnlySpan<char> entity)
+            {
+                // The ulong key is the reversed single-byte character representation of the actual entity string.
+                ulong key = 0;
+                for (int i = 0; i < entity.Length; i++)
+                {
+                    if (entity[i] > 0xFF)
+                    {
+                        return default;
```

Comment:
nit:
```C#
return 0;
```
?

Reply:
Updated. I also changed [L1046](https://github.com/juliushardt/corefx/commit/e2ef1356bd3157a96803ed25d456cd073931b5c9#diff-831332bbb389db07350457a4d92095f8R1046) to `return (char)0;` to make it consistent with [L238](https://github.com/juliushardt/corefx/blob/e2ef1356bd3157a96803ed25d456cd073931b5c9/src/System.Runtime.Extensions/src/System/Net/WebUtility.cs#L238) where it checks `if (entityChar != (char)0)`.